### ISSUE TITLE
Push code for 0.3.0-beta1

### ISSUE THREAD

**Issue by ssgelm** [ravenac95/sudolikeaboss#41]
First of all thank you SO much for the work you've done so far on sudolikeaboss - it's a truly indispensable tool for me.  Given that you seem busy would it be possible to push your code for 0.3.0-beta1 so I and others in the community can update it for the new impending release of 1password?

**Comment by ravenac95** [comment]
@ssgelm Yes! I've been very busy with various things in life for the past several months. I thought I had a branch publicly available. I will check back with you on that. I've been traveling a lot and am only at my work computer (not my personal where I develop sudolikeaboss)

**Comment by ssgelm** [comment]
Maybe I missed it before (or you recently pushed) but I now see the branch `29-fix-for-1password-6-5`.  Thanks!!  I'm going to take a stab at fixing it for 6.8 since that seems to be close to release.

**Comment by smoorez** [comment]
And 6.8 is upon us: https://support.1password.com/kb/201707/

**Comment by Ralnoc** [comment]
@ssgelm 0.3.0-beta1 works fine with 6.8. You just have to run ```sudolikeaboss register``` again.

The real problem is the fact that 1Password will be retiring websocket support in the near future. Once that happens, this will be completely broken. As was mentioned here: [Issue #29 (rudyrichter's comment)](https://github.com/ravenac95/sudolikeaboss/issues/29#issuecomment-315892088) This will have to be solved by changing the underlying method that sudolikeaboss registers with 1Password. 

It will have to be adapted to start using Native Messaging: https://developer.chrome.com/extensions/nativeMessaging

If someone wanted to help @ravenac95 with something, that would be the best area to focus. Attempting to get a version of sudolikeaboss working with native messaging.

**Comment by svoop** [comment]
Would it be possible to make 0.3.0-beta1 available via Homebrew?

It's possible to leave the current stable version untouched but to define either `devel` in the formla which can than be used with `brew install --devel sudolikeaboss`.

**Comment by svoop** [comment]
@ravenac95 As a followup to my previous comment: It should be enough to create a tag `0.3.0beta1` and then extend the homebrew formula something like this (the SHA is missing here of course):

```
require 'formula'

class Sudolikeaboss < Formula
  homepage 'https://github.com/ravenac95/sudolikeaboss'

  if Hardware::CPU.is_64_bit?
    stable do
      version '0.2.1'
      url 'http://dl.bintray.com/ravenac95/sudolikeaboss/sudolikeaboss_0.2.1_darwin_amd64.zip'
      sha256 '23c50e9b6190930f1760122a396a1d6778fbd83266eaf3066623954ec7f0aedd'
    end
    
    devel do
      version '0.3.0beta1'
      url 'http://dl.bintray.com/ravenac95/sudolikeaboss/sudolikeaboss_0.3.0beta1_darwin_amd64.zip'
      sha256 ''
    end
  end
    
  depends_on :arch => :intel

  def install
    bin.install Dir['*']
  end
end
```

(By the way: I've taken the version scheme `0.3.0beta1` from how the formula for Go labels betas.)

**Comment by csawyerYumaed** [comment]
Since 0.2.1 doesn't work at all anymore, with recent versions of 1Password, I don't see why it's not just made the default.

**Comment by svoop** [comment]
Ether way, thanks in advance for cutting a release of `0.3.0beta1`, @ravenac95.

**Comment by n4mine** [comment]
hi, it still doesn't work when lock 1p
osx: 10.12.6
chrome: 60.0.3112.78
sudolikeaboss: 2d9afe1

![img](https://i.1pw.ca/media/native-messaging.png)

**Comment by charliwest** [comment]
On 6.8.1 of 1Password and cannot get this working at all now. The turn off verify option in 1Password seems to be gone now :(

**Comment by brycekahle** [comment]
FYI The code that was in `0.3.0beta1` and some other fixes have been released in my [fork](https://github.com/brycekahle/sudolikeaboss).

**Comment by csawyerYumaed** [comment]
So, there seems to be a few possible workarounds/replacements:

1. Stay on 1Password 6.8 -- and Sierra (I guess High Sierra requires 1p >= 6.8.1?)
2. Use dpaf(https://github.com/tomvachon/dpaf) w/ 1password 6.8.1+ with a 1password subscription and your 1password vault online.
3.  Use iterm2's built-in password manager (set a profile key shortcut to add menu-item 'password manager') -- this uses the macOS Keychain (it creates Login items in the Login keychain under the name 'iTerm2' of kind 'application password').
4. Go back to typing passwords and/or copying and pasting from 1password.
5. Send support requests to annoy 1password into finally waking up, getting a clue and helping @brycekahle or @ravenac95 out.

I'm personally doing 1 and 5. But 1 will eventually stop being an option for me (and most of us) I imagine.

**Comment by svoop** [comment]
> 5. Send support requests to annoy 1password into finally waking up, getting a clue and helping @brycekahle or @ravenac95 out.

Good idea, see [this issue](https://github.com/brycekahle/sudolikeaboss/issues/1) by @brycekahle for the reason why. AgileBits has claimed in the past that 1P customers won't be pushed towards the cloud, so this is sort of a litmus test.
